File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getIntersectionNode(self, headA, headB):
        """
        :type head1, head1: ListNode
        :rtype: ListNode
        """
        temp1 = headA
        temp2 = headB

        while True:
            if temp1 == temp2 :
                return temp1
            if temp1.next is None :
                temp1 = headB
            else :
                temp1 = temp1.next
            if temp2.next is None :
                temp2 = headA
            else :
                temp2 = temp2.next 
            if temp1.next is None and temp2.next is None : break
        if temp1 == temp2 : return temp1
        return None

        

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {

    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        int l1 = 0;
        int l2 = 0 ;
        ListNode temp1 = headA , temp2 = headB ;
        while (temp1 != null){
            l1++;
            temp1 = temp1.next;
        }
        while (temp2 != null){
            l2++;
            temp2 = temp2.next;
        }
        temp1 = headA;
        temp2 = headB ;
        while(l1 > l2) {
            l1--;
            temp1 = temp1.next ;
        }
        while (l2 > l1) {
            l2--;
            temp2 = temp2.next ;
        }
        while(temp1 != null && temp2 != null){
            if(temp1 == temp2) return temp1;
            temp1 = temp1.next ;
            temp2 = temp2.next ;

        }
        return null;
        
    }
}